server-spi-private/pom.xml

```diff
@@ -80,6 +80,17 @@
             <artifactId>hamcrest</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jboss.spec.javax.servlet</groupId>
+            <artifactId>jboss-servlet-api_4.0_spec</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-handler</artifactId>
+            <version>4.1.74.Final</version>
```

Comment:
Keycloak seems to already contain a netty-handler version. It is provided via the `quarkus-vertx` managed dependency.

```
~/dev/playground/keycloak/keycloak-20.0.3 
$ ll lib/lib/main | grep netty-handler
-rw-r--r-- 1 tom tom  526K 12. Jan 17:12 io.netty.netty-handler-4.1.82.Final.jar
-rw-r--r-- 1 tom tom   24K 12. Jan 17:12 io.netty.netty-handler-proxy-4.1.82.Final.jar
```

I suggest to replace the explict netty dependency with the managed `quarkus-vertx` dependency.

Reply:
We should probably avoid adding the SPI provider implementation here and make it specific for Quarkus.

As @thomasdarimont mentioned, we are using Vert.x there already and it is enough to have it implemented only for the Quarkus distribution.